src/cpp/cuda.hpp

```diff
@@ -1671,7 +1671,7 @@ namespace pycuda
     { \
       srcMemoryType = CU_MEMORYTYPE_HOST; \
       py_buffer_wrapper buf_wrapper; \
-      buf_wrapper.get(buf_py.ptr(), PyBUF_ANY_CONTIGUOUS); \
+      buf_wrapper.get(buf_py.ptr(), PyBUF_STRIDED_RO); \
```

Comment:
In the case of memcpy2D and 3D, it seemed reasonable to allow non-contiguous buffers...